zds/utils/templatetags/interventions.py

```diff
@@ -78,10 +79,20 @@ def humane_delta(value):
 
 @register.filter('followed_topics')
 def followed_topics(user):
-    topics_followed = TopicAnswerSubscription.objects.filter(user=user,
-                                                             content_type__model='topic',
-                                                             is_active=True)\
-        .order_by('-last_notification__pubdate')[:10]
+    """
+    gets the ten last followed topics.
+    @TODO : make the "10" parameter a configuration one
+    :param user:
+    :return:
+    """
+    topics_followed = TopicAnswerSubscription.objects\
+        .select_related("content_object")\
+        .annotate(sort_date=F("last_notification__pubdate")
+                  )\
+        .filter(user=user,
+                content_type__model='topic',
+                is_active=True)\
+        .order_by('-sort_date')[:10]
     # This period is a map for link a moment (Today, yesterday, this week, this month, etc.) with
     # the number of days for which we can say we're still in the period
     # for exemple, the tuple (2, 1) means for the period "2" corresponding to "Yesterday" according
```

Comment:
Je sais que c'est out of scope mais tant qu'à toucher cette fonction peux-tu stp remplacer ce bloc de commentaires ?

```diff
-    # This period is a map for link a moment (Today, yesterday, this week, this month, etc.) with
-    # the number of days for which we can say we're still in the period
-    # for exemple, the tuple (2, 1) means for the period "2" corresponding to "Yesterday" according
-    # to humane_delta, means if your pubdate hasn't exceeded one day, we are always at "Yesterday"
-    # Number is use for index for sort map easily
+    # periods is a map associating a period (Today, Yesterday, This week, This month, This year)
+    # with its corresponding number of days: (humane_delta index, number of days).
+    # (3, 7) thus means that passing 3 to humane_delta would return "This week", for which
+    # we'd like pubdate not to exceed 7 days.
```

et éventuellement

```diff
-     periods = ((1, 0), (2, 1), (3, 7), (4, 30), (5, 360))
+     periods = ((1, 0), (2, 1), (3, 7), (4, 30), (5, 365))
```